src/Core/src/Platform/Windows/CheckBoxExtensions.cs

```diff
@@ -1,26 +1,54 @@
 ﻿using Microsoft.Maui.Graphics;
+using Microsoft.UI.Xaml.Controls;
 using Microsoft.UI.Xaml.Media;
 
 namespace Microsoft.Maui.Platform
 {
 	public static class CheckBoxExtensions
 	{
-		public static void UpdateIsChecked(this MauiCheckBox platformCheckBox, ICheckBox check)
+		public static void UpdateIsChecked(this CheckBox platformCheckBox, ICheckBox check)
 		{
 			platformCheckBox.IsChecked = check.IsChecked;
 		}
 
-		public static void UpdateForeground(this MauiCheckBox platformCheckBox, ICheckBox check) 
+		public static void UpdateForeground(this CheckBox platformCheckBox, ICheckBox check) 
 		{
 			var tintBrush = check.Foreground?.ToPlatform();
+			var resources = platformCheckBox.Resources;
 
-			if (tintBrush == null)
+			foreach (string key in _tintColorResourceKeys)
 			{
-				platformCheckBox.TintBrush = new SolidColorBrush(UI.Colors.Black);
-				return;
+				if (tintBrush == null)
+				{
+					resources.Remove(key);
+				}
+				else
+				{
+					resources[key] = tintBrush;
```

Comment:
Does this have any hover color changes - like going lighter or something? Not sure what the differences are, but does the OS have a slightly lighter/transparent color for hover and darker for clicked?

Probably not for this PR so we can keep this one clean, but we probs need to do this at some point. Not API breaking, so we can keep it for RC 2 bug fixes if this is actually needed.

Reply:
I think it behaved nicely and smoothly with mouseover despite me not setting any `*PointerOver` keys. Will double check.